### ISSUE TITLE
[Rx] GrowableTable component

### ISSUE THREAD

**Issue by webinista** [department-of-veterans-affairs/vets-website#2584]
Blocks #2583, #2586 

- Must include a `caption` element for accessibility, that should describe the table's contents. ([Guidance](https://www.w3.org/TR/html5/tabular-data.html#the-caption-element))
- Has wholly different layout at mobile.

#### Desktop view 
- TableRow component #2614
- TableHeader component #2615
- Pagination #2585 

#### Mobile view 
- Not yet defined. Might be able to re-style the Desktop view to accommodate the mobile view.

### Prototypes

- [Desktop](https://marvelapp.com/1h10heg#14227014)
- [Mobile](https://marvelapp.com/iaa9b9#14230873)

**Comment by webinista** [comment]
Mobile and desktop views are quite different. Need to talk through how best to build this component.

One option: 
- Rather than `TableHeader` and `TableRow` components use a list
- Use `display: table-row` for each list item
- Use `display: table-cell` for major child elements, hiding labels on desktop and revealing on mobile.

Another is to conditionally load components based on the viewport's width.

**Comment by akainic** [comment]
This is similar to the `GrowableTable` component in HCA: https://github.com/department-of-veterans-affairs/healthcare-application/blob/master/src/client/components/form-elements/GrowableTable.jsx

**Comment by alexose** [comment]
HCA accomplishes this by using `<div>` elements + the grid system.  This one looks more like a real `<table>` to me, since the data is tabular.  

Factoring in accessibility, I think building a desktop-friendly "table" view and a mobile-friendly "list" view (and loading them conditionally) will probably be the least amount of heartbreak.

One thing I wonder is whether or not it's okay that we lose the sort functionality in the mobile view.  This depends on how many rows we expect the average user has, I guess... anyone have an estimate? @goldenmeanie @angel

**Comment by webinista** [comment]
@alexose I have a vague memory of it not being very hard to restyle a table and its cells by applying `display: block` to td, th, and tr elements. Experiment with that, maybe?

**Comment by alexose** [comment]
I keep forgetting it's 2016 and stuff like that actually works now :)

**Comment by alexose** [comment]
I dropped in [reactable](https://github.com/glittershark/reactable) as a placeholder until we hook this up to the actual API.

**Comment by Drakenhart** [comment]
I was looking into this well, but came across a clash between using tables and styling them versus using div-soup and adding a class to make it read like a table. After a bit of research it seems that you'd want to keep the table a table if at all possible. 

There are many reasons for this. It reduces code that would otherwise need to be added to make divs act like tables for accessibility needs, for example. Screen readers are already programmed in how to read a table.

In other words, why use a crowbar for a hammer?

**Comment by webinista** [comment]
In the [latest mocks](https://marvelapp.com/1h10heg#14227014) the fields have changed a bit.

**Comment by alexose** [comment]
Thanks for the heads up @webinista!

**Comment by webinista** [comment]
Part of #2583

**Comment by U-DON** [comment]
Wasn't quite a GrowableTable since the table used in Rx doesn't have functionality to add, edit, delete, etc. entries in the table. Ended up implementing a SortableTable that sorts its entries based on a click handler for its headers. Closing along with the completion of #2583.
